### PR TITLE
docs: describe new API concepts

### PR DESCRIPTION
**Pull request by eliobischof**
# Which Problems Are Solved

We have not enough internal and external input about our ideas for the new API design. 

# How the Problems Are Solved

We make the concepts easily accessible by publishing them in our docs and making them concise.

# Additional Context

- Contributes to #6305
- Replaces the PR #7821 which defines protos in more detail

Only resources and settings are in the scope of this concept.
A possible solution for defining the outscoped methods could for example look like this:

## ZITADELInsights

query services for auditing, analytics and data synchronization.

- Events
- Milestones

## ZITADELOperations

- Health
- Failed Events
- Views
- Metrics (version, uptime etc.)

### CODE REVIEW

**Comment by eliobischof** [comment]
The PR is in draft until the following is done

- [ ] Add links (new discussion, new issue, ...)
- [ ] Fix example protos (add what's relevant, reduce to the relevant)
- [ ] Replace service lists with actual proto service names
- [ ] Describe the services in more detail where necessary

**Comment by eliobischof** [comment]
Which settings trigger what?

- Settings are generally respected, especially for resources and resource states.
- There are no Session settings, yet, so nothing is respected, but there can be effects if a user is locked for example.
- Users have different settings according to their types.
- Settings are redefined in the new API, so they are easier to understand.

File: docs/docs/apis/v3.mdx [hunk 1]
```diff
@@ -3,13 +3,297 @@ title: APIs V3 (Preview)
...
+## Resource Services
+
+All resource services support the following CRUD operations [as described above](#Standard Resources Behavior).
+
+- Create
+- Read (list, get)
+- Update (partially, success on no changes)
+- Delete
+
+### zitadel.resources.instance.v3alpha.ZITADELInstances
```
[the remark this answers is not in the document]
Comment (eliobischof) [reply docs/docs/apis/v3.mdx:154]: https://protobuf.dev/programming-guides/api/#service-name-globally-unique

File: docs/docs/apis/v3.mdx [hunk 2]
```diff
@@ -3,13 +3,297 @@ title: APIs V3 (Preview)
...
+
+- Limit (partial update of block and audit log retention)
+- BulkLimit (partial update of block and audit log retention for multiple instances)
+
+### zitadel.resources.organization.v3alpha.ZITADELOrganizations
+
+additional to the standard CRUD methods:
+
+- IsUnique
+- GetOIDCInformation
```
[the remark this answers is not in the document]
Comment (eliobischof) [reply docs/docs/apis/v3.mdx:166]: will remove

**Comment by eliobischof** [comment]
The rest we could define like this, but we outscope these services here. 

## ZITADELInsights

query services for auditing, analytics and data synchronization.

- Events
- Milestones

## ZITADELOperations

- Health
- Failed Events
- Views
- Metrics (version, uptime etc.)

File: docs/docs/apis/v3.mdx [hunk 3]
```diff
@@ -3,13 +3,297 @@ title: APIs V3 (Preview)
...
+- Single properties from instance settings are overridable (patchable) in organizations.
+- Some settings support user-defined custom properties that are also overridable in organizations.
+- Improved experience with reusing resources in multiple organizations and instances.
+- Resources are searchable over all organizations with a single call by default.
+
+### HTTP and gRPC Consistency
+
+To make the APIs more consistent and easier to use, we follow the same patterns in all Proto files.
+- Patching is favored over updating resources and settings.
+- HTTP calls are mapped so that query parameters can be used as much as possible. We avoid the annotation `body: "*"`.
```
[the remark this answers is not in the document]
Comment (eliobischof) [reply docs/docs/apis/v3.mdx:59]: I will challenge this externally

**Review by stebenz — APPROVED** [review]

**Pull request merged**